### PR TITLE
Add a simple library for interfacing with message queue (Rabbit MQ via AMQP)

### PR DESCRIPTION
**Pull request by salikh**

### CODE REVIEW

File: go/queue/queue.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	err2 := ch.Connection.Close()
+	if err1 != nil {
+		return err1
+	}
+	return err2
+}
+
+func (ch *Channel) getQueue(queueName string) (amqp.Queue, error) {
+	q, ok := ch.queues[queueName]
+	if !ok {
```
Comment (yunabe) [inline go/queue/queue.go]: early-return to reduce the nest-level.

Maybe, you can write

```golang
if q, ok := ..; ok {
  return q, nil
}
```

File: go/cmd/tool/tool.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+// Binary tool is for quick testing of the queue.
```
Comment (yunabe) [inline go/cmd/tool/tool.go]: go/cmd/tool/tool.go path looks too generic? up to you though.

File: go/go.mod [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module github.com/google/prog-edu-assistant
```
[the remark this answers is not in the document]
Comment (salikh) [reply go/go.mod]: Added. Thanks for catching.

File: go/cmd/tool/tool.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+var commands = map[string]*Cmd{
+	"post":    &Cmd{"Post a message to a queue.", postCommand},
+	"receive": &Cmd{"Receive a single message from a queue.", receiveCommand},
+	"listen":  &Cmd{"Receive messages from a queue forever.", listenCommand},
+}
+
+func main() {
+	flag.Parse()
+	err := run()
```
[the remark this answers is not in the document]
Comment (salikh) [reply go/cmd/tool/tool.go]: Done.

Re: inline run into main:
No, this is on purpose: this relieves me from needing to think whether I need to do log.Fatal(err) or return err.

File: go/cmd/tool/tool.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	if *command == "" {
+		fmt.Println("Error: --command not specified.")
+		fmt.Println("Available commands:")
+		for name, cmd := range commands {
```
[the remark this answers is not in the document]
Comment (salikh) [reply go/cmd/tool/tool.go]: Done.

File: go/queue/queue.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+		false,  // immediate
+		amqp.Publishing{
+			ContentType: "application/octet-stream",
+			Body:        content,
+		})
+}
+
+// Receive returns a (Go) channel that will deliver messages received on the
+// queue specified by a name.
+func (ch *Channel) Receive(queueName string) (<-chan []byte, error) {
```
[the remark this answers is not in the document]
Comment (salikh) [reply go/queue/queue.go]: Thanks for the suggestion. Added.

File: go/queue/queue.go (hunk 1 above)
Comment (salikh) [reply go/queue/queue.go]: Done.

File: go/cmd/tool/tool.go (hunk 2 above)
Comment (salikh) [reply go/cmd/tool/tool.go]: Renamed to go/cmd/try-queue/tryqueue.go

**Pull request merged**